### PR TITLE
Flakify

### PR DESCRIPTION
**Pull request by Lord-Valen**
Adds nix package and devshell for users and contributors respectively.
Also clarifies dependencies in README.

Supersedes #5

### CODE REVIEW

File: nix/watershot/packages.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    src = std.incl self [
+      "Cargo.toml"
+      "Cargo.lock"
+      "src"
+    ];
+
+    cargoLock.lockFile = self + "/Cargo.lock";
+    # TODO: Remove when smithay-client-tooling is on point release
+    cargoLock.outputHashes."smithay-client-tooling-0.16.0" = "sha256-n+s+qH39tna0yN44D6GGlQGZHjsr9FBpp+NZItyqwaE=";
```
Comment (Lord-Valen) [inline nix/watershot/packages.nix:21]: This can be removed when smithay-client-tooling is on a point release.

Comment (Kirottu) [reply nix/watershot/packages.nix:21]: And that is done now, master branch is now using SCTK 0.17.0.

**Comment by Lord-Valen** [comment]
Ready to merge.

**Comment by Kirottu** [comment]
Thank you!

**Comment by n3oney** [comment]
hey @Lord-Valen, I don't really want to open a separate issue for this, but don't you think it'd be nice to include `grim` in the flake and wrap the package to have the binary in PATH?

**Pull request merged**